### PR TITLE
add Lua-cURLv3 luarocks plugin

### PR DESCRIPTION
**Pull request by fiendish**
This is a basic plugin for luarocks/lua-curl in the same style as the other luarocks plugins

### CODE REVIEW

**Comment by fiendish** [comment]
Original commit had (I guess?) wrong hash length for update-package. edited/rebased.

**Comment by tonytheodore** [comment]
Looks good, just a some minor points:
  - use `$(SOURCE_DIR)` instead of `$(1)`
  - owner should generally be yourself
  - patch should be named `lua-curl-1-fixes.patch` and ideally submitted upstream
  - trailing newline at end of file

**Comment by tonytheodore** [comment]
Forgot about the [new style Copyright header](https://github.com/mxe/mxe/pull/1503).

```
This file is part of MXE. See LICENSE.md for licensing information.
```

**Comment by fiendish** [comment]
Can you clarify the owner point?

**Comment by tonytheodore** [comment]
Eventually all packages will use [`$(SOURCE_DIR)`](#1452) and have [consistent patches](#983). Some of this is "mechanical" changes, some not, so it's better if new packages are converted up front. `$(BUILD_DIR)` doesn't seem relevant to `luarocks` (can't build out-of-source?) but `$(SOURCE_DIR)` allows MXE to be used with any source tree i.e. easily test branches in CI systems.

There should only be a single newline at the end of a file - it's a minor nitpick that I perceive as a form of compliment.

The "owner" is ideally someone familiar with the package who can help future MXE maintainers if they have any issues. The patches aren't really appropriate upstream - for some reason I thought the MXE platform was added to luarocks itself upstream.

**Comment by starius** [comment]
@fiendish, could you set owner to URL of your github profile, please?

```mk
$(PKG)_OWNER    := https://github.com/fiendish
```

**Comment by starius** [comment]
Please add a test to `plugins/luarocks/test.lua`

File: plugins/luarocks/lua-curl.mk [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# This file is part of MXE.
+# See index.html for further information.
+
+PKG             := lua-curl
+$(PKG)_WEBSITE  := https://github.com/Lua-cURL/Lua-cURLv3
+$(PKG)_OWNER    := https://github.com/Lua-cURL
+$(PKG)_IGNORE   :=
+$(PKG)_VERSION  := 3735522
```
Comment (starius) [inline plugins/luarocks/lua-curl.mk]: Why the latest version from master branch is used? Does 0.3.4 work?

**Comment by starius** [comment]
I have changed soma files and uploaded new versions to
https://gist.github.com/c3b008930933450da7d40f40aceec269

  * Copyright header
  * Trailing newline
  * Owner
  * version 0.3.4
  * Patch consolidated
  * Test was added. **It does not work for me neither on wine nor on Windows 7 machine.**

**Comment by starius** [comment]
@fiendish 
Could you update this, please?

**Pull request closed without merging**